Documentation/Contributing/UnitTests.md

```diff
@@ -29,22 +27,60 @@ Run the tests on the project located at H:\mrtk.dev, with Unity 2018.4.1f1, outp
+
+To run multiple tests locally, modify your test to call your test multiple times. We have provided a python script to make this more convenient.
+
+First, make sure you have [Python 3.X installed](https://www.python.org/downloads/).
+
+Let's say you have a test that you want to run many times:
 
-## Executing tests on github / CI
-MRTK's CI will build MRTK in all configurations and run all edit and play mode tests. CI can be triggered by posting a
-comment on the github PR `/azp run mrtk_pr` if the user has sufficient rights. CI runs can be seen in the 'checks' tab of the PR.
-Only after all of the tests passed successfully the PR can be merged into mrtk_development.
-> [!NOTE]
-> Some tests will only fail when run from the command line. You can run the tests locally from command line using similar
-setup to what it done in MRTK's CI by running `scripts\test\run_playmode_tests.ps1`
+```
+[UnityTest]
+public IEnumerator MyTest() {...}
+```
 
-## Writing Tests for your code
-To ensure MRTK being a stable and reliable toolkit, every feature should come with unit tests and sample usage in one of the example scenes.
+Run the following from a command line (we recommend [PowerShell](https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell?view=powershell-6#powershell-core))
 
-Preferably when fixing a bug there should also be a test added to avoid running into the same issue again in the future.
+```
+cd scripts\tests
+# Repeat the test 5 times. Default is 100 
+python .\generate_repeat_tests.py -n 5 -t MyTest
+```
 
-Having good test coverage in a big codebase like MRTK is crucial for stability and having confidence when doing changes in code.
+Copy and paste the output into your test file. If you need to run multiple tests in sequence:
 
+```
+cd scripts\tests
+# Repeat the test 5 times. Default is 100 
+python .\generate_repeat_tests.py -n 5 -t MyTest MySecondTest
+```
+
+Your new test file should now contain
+
+```
+[UnityTest]
+public IEnumerator A1MyTest0(){ yield return MyTest();}
+[UnityTest]
+public IEnumerator A2MyTest0(){ yield return MyTest();}
+[UnityTest]
+public IEnumerator A3MyTest0(){ yield return MyTest();}
+[UnityTest]
+public IEnumerator A4MyTest0(){ yield return MyTest();}
+[UnityTest]
+public IEnumerator MyTest() {...}
+```
+
+Open the test runner and you should see your new tests that you can call repeatedly.
+
+## Writing Tests
+To ensure MRTK being a stable and reliable toolkit, every feature should come with unit tests and sample usage in one of the example scenes. Having good test coverage in a big codebase like MRTK is crucial for stability and having confidence when doing changes in code.
```

Comment:
this is explaining our test setup - i would move that to the top of the page bc it's prerequisite for going through your instructions (by then you should already know that we're using unity test runner)